ddtrace/tracer/span.go

```diff
@@ -503,6 +504,7 @@ func (s *span) Finish(opts ...ddtrace.FinishOption) {
 	}
 
 	s.finish(t)
+	orchestrion.GLSPopValue(sharedinternal.ActiveSpanKey)
```

Comment:
What happens if a span finishes before its children?

Reply:
I don't see a case where this could happen in the same goroutine. :thinking: In a multiple goroutine setup where the parent ends in another gouroutine this is a real issue that is not taken care of today. We definitively need to cross goroutine GLS in the future